### ISSUE TITLE
GUI domain system configuration

### ISSUE THREAD

**Issue by marmarek** [QubesOS/qubes-issues#4186]
Configure graphical system in GUI domain (for GUI/Admin domain split), including required qrexec services:
 - [ ] multi-display handling, including dynamic configuration (specifically in fallback case without real GPU passthrough)
 - [x] power reporting (battery, AC)
 - [ ] RPC for managing dom0 powerstate (shutdown, restart, etc.)
 - [ ] screen locker - choose one; if applicable - prevent vt switch
 - [ ] disable "new login" feature
 - [x] Debian packaging for current only dom0 (Fedora) components (e.g. manager, artwork, desktop-linux-xfce4*)
 - [ ] touchpad configuration, and various other input devices configuration that isn't passed through GUI protocol / input proxy

**Comment by xyhhx** [comment]
i just set up sys-gui-gpu (and sys-gui but i dont feel like using that), and noticed a few quirks or would-be nice-to-haves:

- gui domains are based on the default template, which means they don't adopt dom0's desktop environment settings. the user has to configure keyboard shortcuts, panel configuration, etc again.
  - this might be considered a feature, as visual cues make it more obvious which domain is being used (dom0 or gui domain)
  - that said, a user might want to have consistency. perhaps a separate salt formula should handle this to make it opt-in?
- the gui domain can't run as many admin commands as dom0 (for example `qvm-run` doesn't let you use `-u $user` from gui domain)
- running `qubes-sync-appmenus` for _every_ domain is kind of a bummer

i'll jot down more notes as i think of them

**Comment by xyhhx** [comment]
i just set up sys-gui-gpu (and sys-gui but i dont feel like using that), and noticed a few quirks or would-be nice-to-haves:

- gui domains are based on the default template, which means they don't adopt dom0's desktop environment settings. the user has to configure keyboard shortcuts, panel configuration, etc again.
  - this might be considered a feature, as visual cues make it more obvious which domain is being used (dom0 or gui domain)
  - that said, a user might want to have consistency. perhaps a separate salt formula should handle this to make it opt-in?
- the gui domain can't run as many admin commands as dom0 (for example `qvm-run` doesn't let you use `-u $user` from gui domain)
- running `qubes-sync-appmenus` for _every_ domain is kind of a bummer

i'll jot down more notes as i think of them

**Comment by marmarek** [comment]
> * gui domains are based on the default template, which means they don't adopt dom0's desktop environment settings. the user has to configure keyboard shortcuts, panel configuration, etc again.

If you come from customized dom0, yes. But the plan is to have GUI domain from the very start, eventually.

> running `qubes-sync-appmenus` for _every_ domain is kind of a bummer

This is needed only once, though.

> the gui domain can't run as many admin commands as dom0 (for example `qvm-run` doesn't let you use `-u $user` from gui domain)

Right, `-u` doesn't work from non-dom0. But `--service ... qubes.VMRootShell` should (I guess `-u root` could be translated to it automatically).
